sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -639,12 +639,20 @@ def get_tx_power_flag(self):
         '''
         This function returns TX power out of range flag on TX media lane
         '''
-        tx_power_high_alarm_dict = self.xcvr_eeprom.read(consts.TX_POWER_HIGH_ALARM_FLAG)
-        tx_power_low_alarm_dict = self.xcvr_eeprom.read(consts.TX_POWER_LOW_ALARM_FLAG)
-        tx_power_high_warn_dict = self.xcvr_eeprom.read(consts.TX_POWER_HIGH_WARN_FLAG)
-        tx_power_low_warn_dict = self.xcvr_eeprom.read(consts.TX_POWER_LOW_WARN_FLAG)
-        if tx_power_high_alarm_dict is None or tx_power_low_alarm_dict is None or tx_power_high_warn_dict is None or tx_power_low_warn_dict is None:
+        # Read TX power alarm and warning flags in one I2C transaction
+        flags = self.xcvr_eeprom.read(consts.TX_POWER_ALARM_FLAGS_FIELD)
+        if flags is None:
             return None
+        # Extract raw flag dicts
```

Comment:
@jianyuewu There are lot of duplicate code here, can you please combine them 
`def get_alarm_flags(alarm_flag)`: where `alarm_flag` is `TX_POWER`, `RX_POWER`, `TX_BIAS` 

Then use above fn to call individually for get_tx_power_flag(), get_rx_powr_flag() and get_tx_bias_flag()

Reply:
Agree, will update it.